test/ex_doc/retriever_test.exs

```diff
@@ -318,6 +320,76 @@ defmodule ExDoc.RetrieverTest do
     end
   end
 
+  describe "docs_from_modules/2: Erlang" do
+    @describetag :otp23
+
+    test "module", c do
+      erlc(c, :mod, ~S"""
+      %% @doc
+      %% mod docs.
+      -module(mod).
+      -export([function/0]).
+
+      %% @doc
+      %% function/0 docs.
+      function() -> ok.
+      """)
+
+      edoc_to_chunk(:mod)
+
+      [mod] = Retriever.docs_from_modules([:mod], %ExDoc.Config{})
+
+      %ExDoc.ModuleNode{
+        deprecated: nil,
+        doc_line: 0,
+        docs: [function],
+        function_groups: ["Functions"],
+        group: nil,
+        id: "mod",
+        module: :mod,
+        nested_context: nil,
+        nested_title: nil,
+        rendered_doc: nil,
+        source_path: _,
+        source_url: nil,
+        title: "mod",
+        type: :module,
+        typespecs: []
+      } = mod
+
+      assert DocAST.to_string(mod.doc) == "<p>mod docs.</p>"
+
+      %ExDoc.FunctionNode{
+        annotations: [],
+        arity: 0,
+        defaults: [],
+        deprecated: nil,
+        doc_line: 0,
+        group: "Functions",
+        id: "function/0",
+        name: :function,
+        rendered_doc: nil,
+        signature: "function() -> term()\n",
+        source_path: _,
+        source_url: nil,
+        specs: [],
+        type: :function
+      } = function
+
+      assert DocAST.to_string(function.doc) =~ "<p>function/0 docs.</p>"
```

Comment:
for posterity: we use `=~` instead of `==` because depending on version of edoc you might get:

```html
<p>function/0 docs.</p>
```

or:

```html
<a id="function-0"></a><p>function/0 docs.</p>
```

this all might change when the doc chunk work is finalised so let's keep this relaxed.